src/sentry/db/models/utils.py

```diff
@@ -57,10 +57,10 @@ def _resolve(instance, node):
 def slugify_instance(inst, label, reserved=(), max_length=30, *args, **kwargs):
     base_slug = slugify(label)[:max_length]
 
-    if base_slug in reserved:
-        base_slug = None
-    elif base_slug is not None:
```

Comment:
I noticed that this logic would allow bypassing the reserved list if you named your slug, `  admin  `, for example. It skips the check, then strips it, leaving behind a reserved word unchecked.